toolchain/parser/parser_impl.cpp

```diff
@@ -634,20 +714,52 @@ auto ParseTree::Parser::ParseVariableDeclaration() -> Node {
                  /*has_error=*/!pattern || !semi);
 }
 
+auto ParseTree::Parser::ParseInterface() -> Node {
+  // TODO Error handling including the stack thing. Re-read
```

Comment:
Some comment clean-up is needed here and there. Sorry for the not so clean PR. Will do that after work today.